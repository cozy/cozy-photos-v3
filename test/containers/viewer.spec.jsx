'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import Connected, { Viewer } from '../../src/containers/Viewer'

import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
const middlewares = [ thunk ]
const mockStore = configureMockStore(middlewares)

const paramsMock = {
  photoId: '33dda00f0eec15bc3b3c59a615001ac8'
}

const storeMock = mockStore({
  photos: [
    {
      _id: '33dda00f0eec15bc3b3c59a615001ac8'
    },
    {
      _id: '33dda00f0eec15bc3b3c59a615001ac9'
    },
    {
      _id: '33dda00f0eec15bc3b3c59a615001ad0'
    }
  ],
})

describe('Viewer component', () => {
  it('should be displayed correctly with the img src computed from photoId', () => {
    const component = shallow(
      <Connected params={paramsMock} store={storeMock} />
    )
    expect(component.node).toMatchSnapshot()
  })
})
