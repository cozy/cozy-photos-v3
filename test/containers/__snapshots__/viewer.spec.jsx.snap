exports[`Viewer component should be displayed correctly with the img src computed from photoId 1`] = `
<withRouter(Viewer)
  current="33dda00f0eec15bc3b3c59a615001ac8"
  navigateToPhoto={[Function]}
  next="33dda00f0eec15bc3b3c59a615001ac9"
  params={
    Object {
      "photoId": "33dda00f0eec15bc3b3c59a615001ac8",
    }
  }
  previous="33dda00f0eec15bc3b3c59a615001ad0"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;
